EosPayload/lib/config.py

```diff
@@ -0,0 +1,186 @@
+                     with the device_id (padded to 3 digits)
+        """
+        device_id = config.get("device_id")
+        name = config.get("name")
+        return f"{name}-{device_id:03}"
+
+    def configure_device(self, device_config: dict) -> None:
+        device_name = device_config.get("name")
+
+        if device_name is not None and \
+            (not device_name.isascii() or not device_name.replace("-", "").isalnum()):
+            self.logger.error(f"Device name \"{device_name}\" is invalid, falling back to generated name")
+
+        if device_name is None:
+            # Convert default name from CamelCase to lowercase with dashes
+            # Code snippet taken from https://github.com/jpvanhal/inflection
+            device_name = device_config.get("driver_class")
+            device_name = re.sub(r"([A-Z]+)([A-Z][a-z])", r'\1_\2', device_name)
+            device_name = re.sub(r"([a-z\d])([A-Z])", r'\1_\2', device_name)
+            device_name = device_name.replace("_", "-")
+            device_name = device_name.lower()
+
+            device_config.update({"name": device_name})
+            self.logger.info(f"No name provided, generated name {device_name}")
+
+
+        self.logger.info(f"Configuring device \"{device_name}\"")
+
+        if device_config.get("enabled") != "true":
+            self.logger.info(f"{self.config_indent}Driver disabled, skipping")
+            self.disabled_drivers.append(device_name)
+            return
+
+        device_id = device_config.get("device_id")
+        if device_id is None:
+            self.logger.error(f"{self.config_indent}No device ID provided, skipping.")
+            return
+        if device_id in self.used_device_ids:
+            self.logger.error(f"{self.config_indent}Device ID {device_id} already in use, skipping.")
+            return
+        try:
+            device_id = device.Device[device_config["device_id"]].value
+            device_config.update({"device_id": device_id})
+        except KeyError:
+            self.logger.error(f"{self.config_indent}Device ID {device_id} is invalid, skipping.")
+
+        self.logger.info(f"{self.config_indent}Device ID: {device_id}")
+        self.used_device_ids.append(device_id)
+
+        pretty_id = self.get_pretty_id_from_config(device_config)
+        device_config.update({"pretty_id": pretty_id})
+        self.logger.info(f"{self.config_indent}Pretty ID: {pretty_id}")
+
+        driver_class_name = device_config.get("driver_class")
+        self.logger.info(f"{self.config_indent}Driver Class: {driver_class_name}")
+
+        if driver_class_name is None:
+            self.logger.error(f"{self.config_indent}Driver Class is None, skipping")
+            return
+
+        if driver_class_name not in self.valid_driver_classes:
+            self.logger.error(f"{self.config_indent}Driver Class {driver_class_name} is not available, skipping")
```

Comment:
I like available here because it covers both the case where a driver is invalid and the case where it doesn't exist in the first place. Open to other ideas though.

Reply:
how about "Driver Class is invalid or does not exist"?